pages/admin/dashboard/builder.tsx

```diff
@@ -95,16 +97,18 @@ const Builder: Page = () => {
     const router = useRouter();
     const { moduleId } = router.query;
 
-    useEffect(() => {
-        function getUrl() {
-            return `/api/module/${moduleId}`;
-        }
-        async function fetchData() {
-            const response = await axios.get(getUrl());
-            dispatch({ type: "init", payload: response.data });
-        }
-        fetchData();
-    }, [moduleId]);
+    const fetcher = async (url) => {
+        const response = await axios({
+            method: "GET",
+            url,
+        });
+        dispatch({ type: "init", payload: response.data });
```

Comment:
fetcher needs to be in the Builder component so it has access to dispatch. is there a better way to organize this?